x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/monitor_add_page.tsx

```diff
@@ -6,33 +6,54 @@
  */
 
 import React, { useEffect } from 'react';
-import { useDispatch } from 'react-redux';
+import { i18n } from '@kbn/i18n';
+import { useDispatch, useSelector } from 'react-redux';
+import { EuiEmptyPrompt, EuiLoadingLogo } from '@elastic/eui';
 import { useTrackPageview } from '@kbn/observability-plugin/public';
 
-import { getServiceLocations } from '../../state';
+import { getServiceLocations, selectServiceLocationsState } from '../../state';
 import { ServiceAllowedWrapper } from '../common/wrappers/service_allowed_wrapper';
 
 import { useKibanaSpace } from './hooks';
 import { MonitorSteps } from './steps';
 import { MonitorForm } from './form';
+import { LocationsLoadingError } from './locations_loading_error';
 import { ADD_MONITOR_STEPS } from './steps/step_config';
 import { useMonitorAddEditBreadcrumbs } from './use_breadcrumbs';
 
-const MonitorAddPage = () => {
+export const MonitorAddPage = () => {
   useTrackPageview({ app: 'synthetics', path: 'add-monitor' });
   const { space } = useKibanaSpace();
   useTrackPageview({ app: 'synthetics', path: 'add-monitor', delay: 15000 });
   useMonitorAddEditBreadcrumbs();
   const dispatch = useDispatch();
+  const { locationsLoaded, error: locationsError } = useSelector(selectServiceLocationsState);
 
   useEffect(() => {
-    dispatch(getServiceLocations());
-  }, [dispatch]);
+    if (!locationsLoaded) {
+      dispatch(getServiceLocations());
+    }
+  }, [dispatch, locationsLoaded]);
 
-  return (
+  if (locationsError) {
+    return <LocationsLoadingError />;
+  }
+
+  return locationsLoaded ? (
     <MonitorForm space={space?.id}>
       <MonitorSteps stepMap={ADD_MONITOR_STEPS} />
     </MonitorForm>
+  ) : (
+    <EuiEmptyPrompt
```

Comment:
The Edit page actually uses a loader pull from the flyout component I believe? Which is just a spinner. Should I use that instead?

Reply:
I pulled this component from searching "loading" in eui btw.